tests/mod/internal/widget/test_image.cpp

```diff
@@ -264,3 +264,81 @@ RED_AUTO_TEST_CASE(TraceWidgetImage_transparent_png_with_custom_theme_color)
     
     RED_CHECK_SIG(drawable, "\xcb\x73\x41\xea\xa4\x56\x48\x32\xf7\x08\x6d\xc2\x90\x9f\xaa\x99\x68\x8f\xf0\xef");
 }
+
+RED_AUTO_TEST_CASE(TraceWidgetImage_transparent_png_with_custom_theme_color2)
+{
```

Comment:
```cpp
{
  struct Png
  {
    char const* filename;
    array_view sig;
  };

  RED_TEST_CONTEXT_DATA(Png const& png, "filename: " << png.filename, {
    Png{
      FIXTURES_PATH"/logo-redemption-transparent2.png",
      "\x30\xb2\x8b\xc1\x72\x0f\x95\xe4\x24\x06\xeb\x3f\xcd\x4c\x81\x27\x4f\xef\xd9\x9b"_av
    },
    Png{
      FIXTURES_PATH"/logo-redemption-half-transparent.png",
      "\x4b\xaf\x4b\xe9\x01\x75\x89\x13\x44\xf2\xfc\x69\x3b\xdd\xe1\xe3\x80\x06\xe1\x1c"_av
    },
    Png{
      FIXTURES_PATH"/checkerboard-transparent.png",
      "\x2b\x10\x23\x56\xa5\x3e\x0f\xb2\x0a\x5f\x23\x8c\xf6\x1a\xa9\x77\x4f\xa5\x32\xcf"_av
    },
    // ...
  }) {
    .....
    RED_CHECK_SIG_A(d.drawable, png.sig);
  }
}
```

Reply:
Should we rather test expectations one by one? I mean, with your suggestion, if one of these tests fail, we won't know which test will fail exactly

Reply:
it is the purpose of the `TEST_CONTEXT*` to add a message when a test fails. If `CHECK_SIG` fails, there will be

```
... check ... has failed [ sig != "..." ]
Failure occurred in a following context:
    [1/3] filename: ...
```

Reply:
I watched the RED_TEST_CONTEXT_DATA() macro implementation. But in unit testing, I instantiate only one WidgetImage with one .png so it could be a problem for test many .png :/